### PR TITLE
ordered associative task TDobrovolska

### PR DESCRIPTION
**Pull request by TetianaDobrovolska**

### CODE REVIEW

File: ordered_associative_task/student/src/impl.cpp [hunk 1]
```diff
@@ -1 +1,8 @@
 #include "impl.h"
+
+size_t convertMarkTo5PointSystem(size_t markIn100Point)
+{
+    std::map <size_t, size_t> converter {{29, 1}, {49, 2}, {69, 3}, {89, 4}, {90, 5}};
```
Comment (smay1613) [inline ordered_associative_task/student/src/impl.cpp:5]: make it const

File: ordered_associative_task/student/src/impl.cpp [hunk 2]
```diff
@@ -1 +1,8 @@
...
+    auto result = converter.lower_bound(markIn100Point);
+    return converter.end() != result ? result->second : converter.at(90);
```
Comment (smay1613) [inline ordered_associative_task/student/src/impl.cpp:7]: Please avoid this magic 90

File: ordered_associative_task/student/src/impl.cpp [hunk 3]
```diff
@@ -1 +1,8 @@
 #include "impl.h"
+
+size_t convertMarkTo5PointSystem(size_t markIn100Point)
+{
+    std::map <size_t, size_t> converter {{29, 1}, {49, 2}, {69, 3}, {89, 4}, {90, 5}};
+    auto result = converter.lower_bound(markIn100Point);
```
Comment (smay1613) [inline ordered_associative_task/student/src/impl.cpp:6]: const too

File: ordered_associative_task/student/src/impl.cpp (hunk 1 above)
Comment (TetianaDobrovolska) [reply ordered_associative_task/student/src/impl.cpp]: done it

File: ordered_associative_task/student/src/impl.cpp (hunk 2 above)
Comment (TetianaDobrovolska) [reply ordered_associative_task/student/src/impl.cpp]: fixed

File: ordered_associative_task/student/src/impl.cpp (hunk 3 above)
Comment (TetianaDobrovolska) [reply ordered_associative_task/student/src/impl.cpp]: done it

File: ordered_associative_task/student/src/impl.cpp [hunk 4]
```diff
@@ -1 +1,20 @@
 #include "impl.h"
+
+size_t convertMarkTo5PointSystem(size_t markIn100Point)
+{
+    enum Point
```
Comment (smay1613) [inline ordered_associative_task/student/src/impl.cpp:5]: overkill :)

File: ordered_associative_task/student/src/impl.cpp [hunk 5]
```diff
@@ -1 +1,20 @@
...
+        THREE,
+        FOUR,
+        MAX_POINT
+    };
+    const std::map <size_t, size_t> converter {{29, static_cast<size_t>(Point::ONE)},
+                                               {49, static_cast<size_t>(Point::TWO)},
+                                               {69, static_cast<size_t>(Point::THREE)},
+                                               {89, static_cast<size_t>(Point::FOUR)},
+                                               {90, static_cast<size_t>(Point::MAX_POINT)}};
+    const auto result = converter.lower_bound(markIn100Point);
```
Comment (smay1613) [inline ordered_associative_task/student/src/impl.cpp:18]: please revert to previous version and try to use upper_bound

**Review by smay1613 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**